make-uniform/production/Build Standard Database.Rmd

```diff
@@ -129,228 +151,97 @@ dictionary_cat <- dictionary_all %>%
-         survey_tech = survey_CAL_tech
-         ) 
-
-# Clean up
-remove(survey_CAL_key, 
-       survey_CAL_year, 
-       survey_CAL_tech, 
-       dictionary_CAL, 
-       variables_CAL, 
-       survey_CAL, 
-       survey_CAL_select)
-```
-
-
-### Muni
-```{r operator-muni}
-survey_MUN_key  <- 'SF Muni'
-survey_MUN_year <- 2017
-
-# Set survey tech by route if necessary in chunk `combine`
-survey_MUN_tech <- 'local bus'
-
-# Prepare operator-specific dictionary
-dictionary_MUN <- dictionary_all %>%
-  filter(operator == survey_MUN_key)
-
-variables_MUN <- dictionary_MUN$survey_variable %>%
-  unique()
-
-# Read data and select variables of interest
-survey_MUN <- read.csv(f_muni_survey_path, header = TRUE)
-
-# JWH: I don't understand why this code is needed_
-# for (var in variables_MUN) {
-#   if (var %in% colnames(survey_MUN)==FALSE) {
-#     print(paste(var, var %in% colnames(survey_MUN)))
-#   }
-# }
-
-survey_MUN_select <- survey_MUN %>%
-  select(one_of(variables_MUN))
-
-# Reshape the data
-survey_MUN_gather <- survey_MUN_select %>%
-  gather(survey_variable, survey_response, -id) %>%
-  rename(ID = id) %>%
-  mutate(# ID = as.numeric(ID),
-         operator = survey_MUN_key,
-         survey_year = survey_MUN_year,
-         survey_tech = survey_MUN_tech
-         ) 
-
-# Clean up
-remove(survey_MUN_key, survey_MUN_year, survey_MUN_tech, dictionary_MUN, variables_MUN, survey_MUN, survey_MUN_select)
-```
-
-### VTA
-```{r operator-VTA}
-survey_VTA_key  <- 'VTA'
-survey_VTA_year <- 2017
-
-# Set survey tech by route if necessary in chunk `combine`
-survey_VTA_tech <- 'local bus'
-
-# Prepare operator-specific dictionary
-dictionary_VTA <- dictionary_all %>%
-  filter(operator == survey_VTA_key)
-
-variables_VTA <- dictionary_VTA$survey_variable %>%
-  unique()
-
-# Read data and select variables of interest
-survey_VTA <- read.csv(f_vta_survey_path, header = TRUE)
-
-# # JWH: I don't understand why this code is needed_
-# for (var in variables_VTA) {
-#   if (var %in% colnames(survey_VTA)==FALSE) {
-#     print(paste(var, var %in% colnames(survey_VTA)))
-#   }
-# }
-
-survey_VTA_select <- survey_VTA %>%
-  select(one_of(variables_VTA))
-
-# Reshape the data
-survey_VTA_gather <- survey_VTA_select %>%
-  gather(survey_variable, survey_response, -id) %>%
-  rename(ID = id) %>%
-  mutate(ID = as.character(ID),
-         operator = survey_VTA_key,
-         survey_year = survey_VTA_year,
-         survey_tech = survey_VTA_tech
-         ) 
+## Add Surveys
+```{r operator-read-function}
+read_operator <- function(name, year, default_tech, file_path, variable_dictionary) {
+  
+  # name <- 'BART'
+  # year <- 2015
+  # default_tech <- 'heavy rail'
+  # file_path <- f_bart_survey_path
+  # variable_dictionary <- dictionary_all
+  
+  variables_vector <- variable_dictionary %>%
+    filter(operator == name) %>%
+    .$survey_variable %>%
+    unique()
+  
+  df <- read.csv(file_path, header = TRUE, comment.char = "",quote = "\"") %>%
```

Comment:
@jhelsel11: Need to test, but this addresses Issue #14 (I've consolidated the individual reads into a method call)